packages/react/src/components/OutsideClick/index.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+  ComponentProps,
+  ElementType,
+  ReactNode,
+  PropsWithChildren,
+} from 'react';
+import { useOnClickOutside } from '../../hooks/useOnClickOutside';
+
+type NonHaveChildElements =
+  | 'input'
+  | 'textarea'
+  | 'img'
+  | 'br'
+  | 'hr'
+  | 'meta'
+  | 'link'
+  | 'base'
+  | 'col'
+  | 'embed'
+  | 'source'
+  | 'track'
+  | 'wbr';
+
+type NoChildren<Tag extends ElementType> = Tag extends NonHaveChildElements
+  ? { children?: never }
+  : { children?: ReactNode };
```

Comment:
```suggestion
type NoChildren<Tag extends ElementType> = Tag extends NonHaveChildElements
  ? { children?: never }
  : { children: ReactNode }; // (*)
```

기존처럼 children이 포함될 수 있는 태그라면 필수값으로 지정해도 좋을 것 같은데 어떻게 생각하실까요? 🤔

![스크린샷 2024-05-24 오후 4 38 32](https://github.com/modern-agile-team/modern-kit/assets/64779472/f92044c3-1f7f-44a9-af00-cd013ec84053)

Reply:
제가 놓쳤던 부분이었네요.
훨씬 더 좋은 것 같습니다 !
감사합니다 !